Source/OCMock/OCMockObject.m

```diff
@@ -261,50 +284,75 @@ - (void)forwardInvocation:(NSInvocation *)anInvocation
     }
     @catch(NSException *e)
     {
-        [exceptions addObject:e];
+        @synchronized(exceptions)
+        {
+            [exceptions addObject:e];
+        }
         [e raise];
     }
 }
 
 - (BOOL)handleInvocation:(NSInvocation *)anInvocation
 {
-    [invocations addObject:anInvocation];
-
+    @synchronized(invocations)
+    {
+        // When the method has a char* argument we do not retain the arguments. This makes it possible
+        // to match char* args literally and with anyPointer. Not retaining the argument means that
+        // in these cases tests that use their own autorelease pools may fail unexpectedly.
+        if(![anInvocation hasCharPointerArgument])
+            [anInvocation retainArguments];
+        [invocations addObject:anInvocation];
+    }
+    
```

Comment:
Just nil'ing the target doesn't work, it breaks andForwardToRealObject on partial mocks (among other things).  I'm going to see if anything terrible happens if I make a copy of of the invocation to retain arguments on.